code/modules/projectiles/guns/energy/stun.dm

```diff
@@ -41,6 +41,16 @@
 		tesla_zap(src, 3, 10000)
 	qdel(chain)
 
+/obj/item/weapon/gun/energy/shock_revolver/update_icon()

```

Comment:
The existing code works with non-tiling overlays, see: laser gun and captain's laser.
The sprite does have a charge bar looking bit on it that could be easily adapted.